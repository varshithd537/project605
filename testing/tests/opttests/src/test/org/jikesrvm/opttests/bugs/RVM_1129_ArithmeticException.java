/*
 *  This file is part of the Jikes RVM project (http://jikesrvm.org).
 *
 *  This file is licensed to You under the Eclipse Public License (EPL);
 *  You may not use this file except in compliance with the License. You
 *  may obtain a copy of the License at
 *
 *      http://www.opensource.org/licenses/eclipse-1.0.php
 *
 *  See the COPYRIGHT.txt file distributed with this work for information
 *  regarding copyright ownership.
 */
package test.org.jikesrvm.opttests.bugs;

/**
 * Testcase for elision of constructors of subclasses of {@link Throwable}
 * in stack traces. This test case makes sure that the stack trace for
 * {@link ArithmeticException} is the same on all platforms.
 */
public class RVM_1129_ArithmeticException {

  public static void main(String[] args) {
    causeArithmeticException();
  }

  private static void causeArithmeticException() {
    System.out.println(10 / 0);
  }

}
