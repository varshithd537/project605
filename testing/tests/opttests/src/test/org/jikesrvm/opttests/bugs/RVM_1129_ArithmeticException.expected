Exception in thread "MainThread" java.lang.ArithmeticException
   at test.org.jikesrvm.opttests.bugs.RVM_1129_ArithmeticException.causeArithmeticException(RVM_1129_ArithmeticException.java:27)
   at test.org.jikesrvm.opttests.bugs.RVM_1129_ArithmeticException.main(RVM_1129_ArithmeticException.java:23)
